client/src/pages/List.js

```diff
@@ -21,52 +22,23 @@ const useStyles = makeStyles({
   },
 });
 
-const taskTypes = {
-  "task-1": { id: "task-1", title: "art", status: "red", note: "March 14" },
-  "task-2": { id: "task-2", title: "philosophy", status: "red" },
-  "task-3": { id: "task-3", title: "cooking", status: "red" },
-  "task-4": { id: "task-4", title: "gym", status: "red" },
-  "task-5": { id: "task-5", title: "math", status: "green" },
-  "task-6": { id: "task-6", title: "english", status: "green" },
-  "task-7": { id: "task-7", title: "science", status: "green" },
-  "task-8": { id: "task-8", title: "history", status: "green" },
-};
-
-const columnTypes = {
-  "col-1": {
-    id: "col-1",
-    name: "To do",
-    taskIds: [...Object.keys(taskTypes)],
-  },
-  "col-2": {
-    id: "col-2",
-    name: "In progress",
-    taskIds: [],
-  },
-  "col-3": {
-    id: "col-3",
-    name: "Review",
-    taskIds: [],
-  },
-  "col-4": {
-    id: "col-4",
-    name: "Completed",
-    taskIds: [],
-  },
-};
-
-const initialData = {
-  columns: columnTypes,
-  tasks: taskTypes,
-  columnOrder: [...Object.keys(columnTypes)],
-};
-
-function List() {
+function List(props) {
   const classes = useStyles();
-  const [data, setData] = useState(initialData);
+  const { loadedData, currBoardId } = props;
+  const [boardData, setBoardData] = useState(loadedData);
+  useEffect(() => {
+    setBoardData(loadedData);
+  }, [loadedData]);
+
+  function handleCardMove(data, movement, newState) {
+    data.movement = movement;
+    setBoardData(newState);
+    editBoard(currBoardId, data);
```

Comment:
I mentioned that the `currBoardId` is undefined in another comment already, but also I'm not sure if the back-end will persist changes when the column order is changed by drag and drop. Maybe it is because the `newState` is not being sent to the server?